structuralalignment/superposition/theseus.py

```diff
@@ -0,0 +1,200 @@
+import atomium
+import subprocess
+from base import CommandLineWrapper
+from structuralalignment.utils import enter_temp_directory
+
+
+"""
+Theseus superposes a set of macromolecular structures simultaneously
+using the method of maximum like-lihood (ML), rather than the
+conventional least-squares criterion.Theseus assumes that the
+structures are distributed according to a matrix Gaussian
+distribution and that the eigenvalues of the atomic
+covariancematrix are hierarchically distributed according
+to an inverse gamma distribution. This ML superpositioning model
+produces much more accurate results by essentially downweighting
+variable regions of the structuresand by correcting for correlations
+among atoms.
+"""
+
+
+class Theseus(CommandLineWrapper):
+    """
+    A mode for superpositioning macromolecules with
+    identical  sequences  and  numbers  ofresidues,
+    for instance, multiple models in an NMR family
+    or multiple structures from differentcrystal
+    forms of the same protein.
+
+    Parameter
+    ---------
+    CommandLineWrapper : class
+
+    """
+
+    def _calculate(self, structures, **kwargs):
+        """
+        Superpose structures with identical sequences with the tool Theseus
+
+        Parameter
+        ---------
+        structures : list
+            list of atomium objects
+
+        **kwargs : dictionary
+            optional parameter
+        """
+        fetched_pdbs = []
+        fetched_pdbs_filename = []
+
+        for pdb in structures:
+            fetched_pdbs.append(pdb)
+        with enter_temp_directory():
+            for pdb in fetched_pdbs:
+                pdb_filename = pdb.code + "." + pdb.filetype
+                pdb.model.save(pdb_filename)
+                fetched_pdbs_filename.append(pdb_filename)
+
+            output = subprocess.check_output(["theseus"] + fetched_pdbs_filename)
+
+        return self._parse(output)
+
+    def _parse(self, output):
+        """
+        Parse the output
+        """
+        return output
+
+
+class TheseusAlign(CommandLineWrapper):
+    """
+    Superpose structures with different sequences but similar structures
+
+    An alignment mode for superpositioning structures with different
+    sequences, for example, multiple structures of the cytochrome c
+    protein from different species or multiple mutated structures of
+    hen egg white lysozyme.
+    """
+
+
+with enter_temp_directory():
```

Comment:
This line doesn't belong here, I think?

Reply:
my idea was to enter one temp directory instead of multiple

Reply:
I see what you mean, but it won't work like that:

* Indentation level doesn't match the class level
* This will be called upon class definition, not object instantiation
* Generally we don't want code outside the methods; if anything, definition of [class attributes](https://stackoverflow.com/questions/207000/what-is-the-difference-between-class-and-instance-attributes)

I wouldn't worry about performance here. Additionally, it's useful to have independent workspaces for each run in case they fail (so we can debug); temporary directories will be eventually deleted anyway!